### PR TITLE
Tools improvements

### PR DESCRIPTION
**Pull request by d2s**
## What is the goal of this PR?

- Find out potential code reliability issues faster
- Improve consistency of future improvements
- Improve documentation

### chore: add EditorConfig settings

- Add EditorConfig settings

#### Reason for adding

"EditorConfig helps maintain consistent coding styles
for multiple developers working on the same project
across various editors and IDEs."

#### More details

- <https://editorconfig.org>

### docs: improve Markdown structure

- Improve Markdown structure
- Solve `markdownlint` warnings
  - `MD040/fenced-code-language`: Fenced code blocks should have a language specified
- Add inline code highlights

### chore: improve .gitignore patterns

- Improve `.gitignore` patterns

### chore: add ESLint & Prettier tooling

- Add ESLint & Prettier tooling
- Add ESLint configuration that is compatible with Prettier
- Add npm run tasks for running the tools
  - `npm run eslint`
    - Checks files under `bin/` directory for structure issues
  - `npm run prettier`
    - Autoformats code style for the files
- Update `package-lock.json` lockfile

#### Reason for adding

- Find out potential code reliability issues faster
- Improve consistency of future improvements

#### Command used for adding the dependencies

```sh
npm install --save-dev --save-exact eslint eslint-config-prettier eslint-plugin-prettier
npm install --save-dev --save-exact prettier
```

#### More details

- <https://prettier.io>
- <https://eslint.org>

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    es6: true,
+    node: true,
+  },
+  globals: {
+    Atomics: "readonly",
+    SharedArrayBuffer: "readonly",
+  },
+  parserOptions: {
+    ecmaVersion: 2018,
+    sourceType: "module",
```
Comment (Mulperi) [inline .eslintrc.js]: Should this actually be "script"?

File: .gitignore [hunk 2]
```diff
@@ -1 +1,26 @@
...
+dist/
+build/
+
+# Cached files
+.eslintcache
+
+# Local package management cache
 node_modules/
+
+# Ignore Yarn lockfile because we you npm
```
Comment (Mulperi) [inline .gitignore]: we you?

File: .eslintrc.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+module.exports = {
+  env: {
+    browser: false,
+    es6: true,
+    node: true,
+  },
+  globals: {
+    Atomics: "readonly",
+    SharedArrayBuffer: "readonly",
```
Comment (Mulperi) [inline .eslintrc.js]: Do we need these?

Comment (d2s) [reply .eslintrc.js]: Most likely not. Removed those lines...

**Comment by d2s** [comment]
Updated the pull request with the requested changes.

**Review by Mulperi — APPROVED** [review]

**Pull request merged**